libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/utils/DirectionsRouteMissingConditionsCheck.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.navigation.base.internal.utils
+
+import androidx.annotation.VisibleForTesting
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+
+internal object DirectionsRouteMissingConditionsCheck {
+
+    private val exclusiveQueries: Map<String, String> = QueriesProvider.provideQueries()
+
+    @VisibleForTesting
+    internal val ERROR_MESSAGE_TEMPLATE =
+        """DirectionsRoute doesn't contain enough data for turn by turn navigation. 
+                The SDK must consume directly (without mappers) NavigationRoute.
+                See `MapboxNavigation#requestRoutes(RouteOptions, NavigationRouterCallback)` 
+                and `NavigationRoute#create`). Params cannot be processed with DirectionsRoute: """
+            .trimIndent()
+
+    /**
+     * Some **new** Navigation API features brings data with [DirectionsResponse] that requires for
```

Comment:
is required

Reply:
what do you mean?

Reply:
"data is required for", not "data requires for".